streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -204,12 +221,11 @@ void unlock(final TaskId taskId) throws IOException {
      * @param cleanupDelayMs only remove directories if they haven't been modified for at least
      *                       this amount of time (milliseconds)
      */
-    public void cleanRemovedTasks(final long cleanupDelayMs) {
+    public synchronized void cleanRemovedTasks(final long cleanupDelayMs) {
```

Comment:
As we schedule this in an own thread and remove from main poll-loop, I think we don't need parameter `cleanupDelayMs` anymore.